pkg/controller/directvolumemigration/rsync.go

```diff
@@ -1244,62 +1287,57 @@ func getMD5Hash(s string) string {
+func (t *Task) hasAllProgressReportingCompleted() (bool, error) {
 	t.Owner.Status.RunningPods = []*migapi.PodProgress{}
 	t.Owner.Status.FailedPods = []*migapi.PodProgress{}
 	t.Owner.Status.SuccessfulPods = []*migapi.PodProgress{}
 	t.Owner.Status.PendingPods = []*migapi.PodProgress{}
 	var pendingPods []string
-	var runningPods []string
 	pvcMap := t.getPVCNamespaceMap()
 	for ns, vols := range pvcMap {
 		for _, vol := range vols {
+			operation := t.Owner.Status.GetRsyncOperationStatusForPVC(&corev1.ObjectReference{
+				Namespace: ns,
+				Name:      vol.Name,
+			})
 			dvmp := migapi.DirectVolumeMigrationProgress{}
 			err := t.Client.Get(context.TODO(), types.NamespacedName{
 				Name:      getMD5Hash(t.Owner.Name + vol.Name + ns),
 				Namespace: migapi.OpenshiftMigrationNamespace,
 			}, &dvmp)
 			if err != nil {
-				// todo, need to start thinking about collecting this error and reporting other CR's progress
-				return false, false, err
+				return false, err
 			}
-			objRef := &corev1.ObjectReference{
-				Namespace: ns,
-				Name:      fmt.Sprintf("directvolumemigration-rsync-transfer-%s", vol.Name),
+			podProgress := &migapi.PodProgress{
+				ObjectReference: &corev1.ObjectReference{
+					Namespace: ns,
+					Name:      dvmp.Status.PodName,
+				},
+				PVCReference: &corev1.ObjectReference{
+					Namespace: ns,
+					Name:      vol.Name,
+				},
+				LastObservedProgressPercent: dvmp.Status.TotalProgressPercentage,
+				LastObservedTransferRate:    dvmp.Status.LastObservedTransferRate,
+				TotalElapsedTime:            dvmp.Status.RsyncElapsedTime,
 			}
 			switch {
-			case dvmp.Status.PodPhase == corev1.PodRunning:
-				t.Owner.Status.RunningPods = append(t.Owner.Status.RunningPods, &migapi.PodProgress{
-					ObjectReference:             objRef,
-					LastObservedProgressPercent: dvmp.Status.LastObservedProgressPercent,
-					LastObservedTransferRate:    dvmp.Status.LastObservedTransferRate,
-				})
-				runningPods = append(runningPods, fmt.Sprintf("%s/%s", objRef.Name, objRef.Namespace))
-			case dvmp.Status.PodPhase == corev1.PodFailed:
-				t.Owner.Status.FailedPods = append(t.Owner.Status.FailedPods, &migapi.PodProgress{
-					ObjectReference:             objRef,
-					LastObservedProgressPercent: dvmp.Status.LastObservedProgressPercent,
-					LastObservedTransferRate:    dvmp.Status.LastObservedTransferRate,
-				})
-			case dvmp.Status.PodPhase == corev1.PodSucceeded:
-				t.Owner.Status.SuccessfulPods = append(t.Owner.Status.SuccessfulPods, &migapi.PodProgress{
-					ObjectReference:             objRef,
-					LastObservedProgressPercent: dvmp.Status.LastObservedProgressPercent,
-					LastObservedTransferRate:    dvmp.Status.LastObservedTransferRate,
-				})
+			case dvmp.Status.PodPhase == corev1.PodRunning ||
+				// when last pod failed, but whole operation is not failed, show as running until next pod is created to avoid flicker
+				(dvmp.Status.PodPhase == corev1.PodFailed && !operation.Failed):
```

Comment:
this flickering is dubious. A smart user sees in the dvm CR that the pod is running but they see in dvmp CR that pod is failing? isn't that contradictory?

Reply:
@alaypatel07 it happens for a very small period of time but the end effect of allowing that behaviour is bad in the UI. Like the pod goes completely missing from the details page and then suddenly shows up with a new Pod name because it got retried. I wouldn't worry about the CLI user as this happens for a very small amount of time (time between a failed pod being registered as failed to the new one being created, a few ms)

Reply:
@pranavgaikwad why dont we have `backoffLimit` on `dvmp.spec` and let it declare failure after it sees n pods instead of polluting the logic here?

Reply:
@alaypatel07 I don't think it's DVMP's concern to even know about backOffLimit, if you put backOffLimit in DVMP it will also have to run to Completion. You cannot re-use the same CR over and over again.

Reply:
> @alaypatel07 it happens for a very small period of time but the end effect of allowing that behaviour is bad in the UI. Like the pod goes completely missing from the details page and then suddenly shows up with a new Pod name because it got retried. I wouldn't worry about the CLI user as this happens for a very small amount of time (time between a failed pod being registered as failed to the new one being created, a few ms) But that flicker is pretty evident in the UI if youre continuously looking at the screen. I hope I explained it right, but I can show you the video if not clear.

Ack, if we are lucky it will not be visible to the user. But I still feel this switch cases are little over loaded. We can delegate some of the work from here to dvmp controller and have this cleanly depend on the dvmp status purely. If dvmp status says we succeeded, failed or are running, dvm blindly copies it. its a good way to hide the complexity underneath the API status IMO

Reply:
Synced with Pranav over a call, this is not an issue as of now, we will resolve and re-visit when the time is right